cmd/cli/app/repo/repo_register.go

```diff
@@ -67,6 +66,25 @@ func RegisterCmd(ctx context.Context, cmd *cobra.Command, conn *grpc.ClientConn)
 		return cli.MessageAndError("Error getting list of repos", err)
 	}
 
+	alreadyRegisteredReposSet := make(map[string]bool)
```

Comment:
We're already using [`k8s.io/apimachinery/pkg/util/sets.Set`](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#Set) elsewhere in the code.  With that, this would be:

With that, you can use `alreadyRegisteredReposSet.HasAny(inputRepositories)` or `.Intersection(inputRepositories)` to find the overlapping items.

In fact, you may be able to use `Results` as a set item, though I suspect there's something in the proto definition to prevent the copying needed...

Reply:
I changed to [`k8s.io/apimachinery/pkg/util/sets.Set`](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#Set).

> With that, you can use alreadyRegisteredReposSet.HasAny(inputRepositories) or .Intersection(inputRepositories) to find the overlapping items.

Functional programming style would be more precise, but here, the sets are quite different. To do set differences and stuff like that, we would have to convert set types to be the same and create redundant collections. This is fine IMO.

> though I suspect there's something in the proto definition to prevent the copying needed

No, Results is comparable, so it would work as a key, but it's not worth storing everything, we would still need a string set to compare with input.